modules/core/test/test_io.cpp

```diff
@@ -740,3 +740,25 @@ TEST(Core_InputOutput, filestorage_xml_base64)
 {
     CV_Base64IOTest test("base64_test_tmp_file.xml"); test.safe_run();
 }
+
+TEST(Core_InputOutput, filestorage_yml_vec2i)
+{
+    const std::string file_name = "vec2i.yml";
+    cv::Vec2i vec(2, 1), ovec;
+
+    /* write */
+    {
+        cv::FileStorage fs(file_name, cv::FileStorage::WRITE);
+        fs << "prms0" << "{" << "vec0" << vec << "}";
+        fs.release();
+    }
+
+    /* read */
+    {
+        cv::FileStorage fs(file_name, cv::FileStorage::READ);
+        fs["prms0"]["vec0"] >> ovec;
+    }
+
+    EXPECT_EQ(vec(0), ovec(0));
```

Comment:
the file should probably be removed in the end, or else it should be a temporary file, or else you can use memory-based file storage.